### ISSUE TITLE
Built-in scheduled jobs need configurable jitter/randomization

### ISSUE THREAD

**Issue by jcmcken** [saltstack/salt#46462]
### Description of Issue/Question

Built-in scheduled jobs (e.g. ``__update_grains``) need configurable jitter/randomization to avoid creating a thundering herd against the Salt master. 

There seem to be certain configuration settings to make the minions themselves delay initial communications to the master on startup in various situations, but it's not clear that this affects scheduled jobs or anything else that occurs after the minion has already started up.

I think there might be away to work around this problem by overriding ``__update_grains`` and configuring ``splay``, but this seems like an un-safe operation. Is ``__update_grains`` considered part of Salt's "internals"? Will this suddenly change between releases without notice? If such a workaround exists, can it be documented?

### Steps to Reproduce Issue

* Start up a couple dozen minions with ``grains_refresh_every`` Minion option set to e.g. ``5`` (refresh grains every minute).
* Monitor the Salt master's CPU, and notice that it spikes heavily exactly every 5 minutes.

### Versions Report

```
Salt Version:
           Salt: 2017.7.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.17.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by garethgreenaway** [comment]
@jcmcken Thanks for the report.  This would be a good addition to include in the scheduler and should be fairly easy to add.  Off the top of my head I think the best approach would be to have a global `splay` option that would affect all jobs.

**Comment by rallytime** [comment]
Closed via #46525
